packages/packages.go

```diff
@@ -231,6 +276,9 @@ func (i *FileSystemIndexer) getPackagesFromFileSystem(ctx context.Context) (Pack
 	span.Context.SetLabel("indexer", i.label)
 	defer span.End()
 
+	i.m.RLock()
+	defer i.m.RUnlock()
+
```

Comment:
Not sure this method requires a lock.
This method is used by Init() and by the new method to update the package list `watchPackageFileSystem`. IIUC this usage would be more related to a writer lock, since it builds the new package list.

However, I think it would be needed a writer lock inside the `watchPackageFileSystem` that it is already present to update the lists.